applications/plugins/SofaPython/Binding_BoundingBoxData.cpp

```diff
@@ -0,0 +1,70 @@
+#include "Binding_BoundingBoxData.h"
+#include "Binding_Data.h"
+#include "PythonToSofa.inl"
+
+using sofa::defaulttype::BoundingBox;
+using namespace sofa::core::objectmodel;
+
+static inline Data<BoundingBox>* get_boundingbox(PyObject* obj) {
+    return sofa::py::unwrap<Data<sofa::defaulttype::BoundingBox> >(obj);
+}
+
+
+SP_CLASS_ATTR_GET(BoundingBox, minBBox)(PyObject *self, void*)
+{
+    Data<BoundingBox>* obj = get_boundingbox( self );
+    if (!obj)
+    {
+        PyErr_BadArgument();
+        return NULL;
+    }
+
+    PyObject* shape = PyList_New(3);
+    PyList_SetItem(shape, 0, PyFloat_FromDouble(obj->getValue().minBBox().x()));
+    PyList_SetItem(shape, 1, PyFloat_FromDouble(obj->getValue().minBBox().y()));
+    PyList_SetItem(shape, 2, PyFloat_FromDouble(obj->getValue().minBBox().z()));
+
+    return shape;
+}
+
+SP_CLASS_ATTR_GET(BoundingBox, maxBBox)(PyObject *self, void*)
+{
+    Data<BoundingBox>* obj = get_boundingbox( self );
+    if (!obj)
+    {
+        PyErr_BadArgument();
+        return NULL;
+    }
+
+    PyObject* shape = PyList_New(3);
+    PyList_SetItem(shape, 0, PyFloat_FromDouble(obj->getValue().maxBBox().x()));
+    PyList_SetItem(shape, 1, PyFloat_FromDouble(obj->getValue().maxBBox().y()));
+    PyList_SetItem(shape, 2, PyFloat_FromDouble(obj->getValue().maxBBox().z()));
+
+    return shape;
+}
+
+SP_CLASS_ATTR_SET(BoundingBox, minBBox)(PyObject */*self*/, PyObject * /*args*/, void*)
+{
+    SP_MESSAGE_ERROR("BoundingBox attributes are read-only")
```

Comment:
What is -1 for ? Is the the standard way to return a problem in a setter ? 
If this is the case I think there is no need to send a message in addition to PyErro_BadArgument. 
And if you want more details in the exception risen then it is best to use PyErr_SetString to set the message associated with the exxception. 

In case the exception is not caught by python the message will be finally be printed by sofa using the sofa msg api.

Reply:
I got inspiration from DataFileName, that returns -1.
I will adjust

Reply:
I don't know if it is the standard way or the best way to do it, but that's how it is in every SP_CLASS_ATTR_SET methods that I found in Sofa.. :/